features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+      }
+      if (!needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign two transactions one to approve synthetic ${tokenSymbol} and a second to preform the liquidation.`;
+      }
+      if (needCollateralAllowance() && needTokenAllowance()) {
+        return `You will need to sign three transactions one to approve synthetic ${tokenSymbol}, one to approve collateral ${collSymbol} and a third to preform the liquidation.`;
+      }
+    };
+
+    const executeDeposit = async () => {
+      if (collateralToDeposit && !collBalanceTooLow() && emp) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+        const collateralToDepositWei = toWei(collateralToDeposit);
+
+        try {
+          if (needCollateralAllowance()) await setMaxCollateralAllowance();
+          const tx = await emp.depositTo(props.selectedSponsor, [
+            collateralToDepositWei,
+          ]);
+          setHash(tx.hash as string);
+          await tx.wait();
+
+          setSuccess(true);
+        } catch (error) {
+          console.error(error);
+          setError(error);
+        }
+      } else {
+        setError(new Error("Please check that you are connected."));
+      }
+    };
+
+    const executeLiquidation = async () => {
+      if (
+        minCollPerToken &&
+        maxCollPerToken &&
+        maxTokensToLiquidate &&
+        deadline &&
+        !collBalanceTooLow() &&
+        !tokenBalanceTooLow &&
+        emp
+      ) {
+        setHash(null);
+        setSuccess(null);
+        setError(null);
+        const minCollPerTokenWei = toWei(minCollPerToken);
+        const maxCollPerTokenWei = toWei(maxCollPerToken);
+        const maxTokensToLiquidateWei = toWei(maxTokensToLiquidate);
+        const deadlineTimestamp = Math.floor(Date.now() / 1000) + deadline;
+        try {
+          if (needCollateralAllowance()) await setMaxCollateralAllowance();
+          if (needTokenAllowance()) await setMaxTokenAllowance();
+
+          const tx = await emp.createLiquidation(
+            props.selectedSponsor,
+            [minCollPerTokenWei],
+            [maxCollPerTokenWei],
+            [maxTokensToLiquidateWei],
+            deadlineTimestamp
+          );
+          setHash(tx.hash as string);
+          await tx.wait();
+
+          setSuccess(true);
+        } catch (error) {
+          console.error(error);
+          setError(error);
+        }
+      } else {
+        setError(new Error("Please check that you are connected."));
+      }
+    };
+
+    return (
+      <Dialog open={props.isDialogShowing} onClose={props.handleClose}>
+        <PositionDialog>
+          <Container>
+            <Box>
+              <Box>
+                <h1>Additional Position Actions</h1>
+                <Status>
+                  <Label>Sponsor: </Label>
+                  <a
+                    href={
+                      props.selectedSponsor
```

Comment:
So if you catch `props.selectedSponsor !== undefined` higher up then you won't need these conditionals in the React elements which is easier to maintain going forwards